### PR TITLE
[RFR][1LP] Linty

### PR DESCRIPTION
**Pull request by psav**
This will never pass green
Lots of lint fixes

### CODE REVIEW

**Comment by dajoRH** [comment]
Would you mind rebasing this Pull Request against latest master, please? :trollface:
*CFME QE Bot*

**Comment by mshriver** [comment]
This brought a tear to my eye.

File: cfme/common/vm.py [hunk 1]
```diff
@@ -495,15 +495,15 @@ def set_ownership(self, user=None, group=None, click_cancel=False, click_reset=F
         cfg_btn('Set Ownership')
         if click_reset:
             action = form_buttons.reset
-            msg_assert = lambda: flash.assert_message_match(
```
Comment (RonnyPfannschmidt) [inline cfme/common/vm.py]: the switch to partial needs a realignment of the next line to fit the requirements

File: cfme/configure/red_hat_updates.py [hunk 2]
```diff
@@ -90,10 +90,11 @@ def make_update_button(text):
 }
 
 
-appliances_table = lambda: version.pick({
-    version.LOWEST: CheckboxTable("//div[@id='form_div']/table[@class='style3']"),
-    '5.4': CheckboxTable("//div[@id='form_div']/table")
-})
+def appliances_table():
```
Comment (RonnyPfannschmidt) [inline cfme/configure/red_hat_updates.py]: im starting to see more and more of those detached version picks, i do wonder if a abstraction would help

File: cfme/control/explorer.py [hunk 3]
```diff
@@ -1226,6 +1228,10 @@ def _fill(self):
                 raise TypeError("evaluate must be either string with dropdown value or function "
                                 "or a 2-tuple with (dropdown value, dict of values to set)")
         fill(self.form, fill_details)
+        if form_func is None:
```
Comment (RonnyPfannschmidt) [inline cfme/control/explorer.py]: please just make the call optional, its more clear and saves a dummy function/variable

File: cfme/fixtures/pytest_selenium.py [hunk 4]
```diff
@@ -1168,8 +1169,8 @@ def _login_func():
             recycle = True
         # If the page is blocked, then recycle...
         if (
-                is_displayed("//div[@id='blocker_div' or @id='notification']", _no_deeper=True)
-                or is_displayed(".modal-backdrop.fade.in", _no_deeper=True)):
+                is_displayed("//div[@id='blocker_div' or @id='notification']", _no_deeper=True) or
```
Comment (RonnyPfannschmidt) [inline cfme/fixtures/pytest_selenium.py]: im starting to wonder if a any_displayed helper might make sense

File: cfme/infrastructure/cluster.py [hunk 5]
```diff
@@ -49,8 +49,8 @@ def __init__(self, name, provider, appliance=None):
         self.quad_name = 'cluster'
 
         col = rest_api().collections
-        self._id = [cl.id for cl in col.clusters.all if cl.name == self._short_name
-                    and cl.ems_id == self.provider.id][-1]
+        self._id = [cl.id for cl in col.clusters.all if cl.name == self._short_name and
```
Comment (RonnyPfannschmidt) [inline cfme/infrastructure/cluster.py]: a more nice formatting for multiline conditional list expression is newline before expression * for and another newline before the condition, so one line will be the loop, and another line will be the condition

btw, why do we obtain the last element instead of the first matching?

if it its ok to take the first one, we could use next + a generator expression

File: cfme/web_ui/menu.py [hunk 6]
```diff
@@ -493,7 +493,8 @@ def f(_):
                 el = "{} | {}".format(active_loc, inactive_loc)
                 cls._try_nav(el, toplevel, secondlevel)
 
-            nav_fn = lambda: cls._nav_to_fn(toplevel, secondlevel, reset_action, _final=True)
+            def nav_fn():
```
Comment (RonnyPfannschmidt) [inline cfme/web_ui/menu.py]: this one and the next could potentially be partials as well, i think its a case where we should use a partial

File: utils/testgen.py [hunk 7]
```diff
@@ -181,7 +181,8 @@ def fixture_filter(metafunc, argnames, argvalues):
     keep_index = [e[0] for e in enumerate(argnames) if e[1] in metafunc.fixturenames]
```
Comment (RonnyPfannschmidt) [inline utils/testgen.py]: future todo? we should use index, name here, and generate a itemgetter to be applied to the list instead of just a normal list expression function

File: cfme/configure/red_hat_updates.py (hunk 2 above)
Comment (psav) [inline cfme/configure/red_hat_updates.py]: They will go away in widgetastic I am almost positive

File: cfme/control/explorer.py (hunk 3 above)
Comment (psav) [inline cfme/control/explorer.py]: What do you mean make it optional?

File: cfme/fixtures/pytest_selenium.py (hunk 4 above)
Comment (psav) [inline cfme/fixtures/pytest_selenium.py]: Not part of this PR

File: cfme/web_ui/menu.py (hunk 6 above)
Comment (psav) [inline cfme/web_ui/menu.py]: OK, I considered it...but didn't
will change

File: utils/testgen.py (hunk 7 above)
Comment (psav) [inline utils/testgen.py]: sure, another pr

File: cfme/control/explorer.py (hunk 3 above)
Comment (RonnyPfannschmidt) [inline cfme/control/explorer.py]: instead of declaring the missing function and then calling it - just dont call it, if it is missing

File: cfme/fixtures/pytest_selenium.py (hunk 4 above)
Comment (mfalesni) [inline cfme/fixtures/pytest_selenium.py]: @RonnyPfannschmidt This is much faster and atomic. Only one selenium call instead of two.

File: cfme/web_ui/menu.py (hunk 6 above)
Comment (psav) [inline cfme/web_ui/menu.py]: I won't change this - it will be gone soon

**Comment by dajoRH** [comment]
Would you mind rebasing this Pull Request against latest master, please? :trollface:
*CFME QE Bot*

File: cfme/fixtures/pytest_selenium.py (hunk 4 above)
Comment (RonnyPfannschmidt) [inline cfme/fixtures/pytest_selenium.py]: @mfalesni we do have an or there with a second is_displayed, perhaps a better abstraction level would be xpath expressions, however that is a reasonably large can of worms we shouldn't touch all too early

lets make a future note perhaps
